Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp

```diff
@@ -1478,24 +1481,23 @@ void TerrainSystem::OnTick(float /*deltaTime*/, AZ::ScriptTimePoint /*time*/)
 
         // This needs to happen before the "system active" check below, because activating the system will cause the various
         // terrain layer areas to request the current world bounds.
-        if (m_requestedSettings.m_worldBounds != m_currentSettings.m_worldBounds)
+        if (m_requestedSettings.m_heightRange != m_requestedSettings.m_heightRange)
         {
-            m_dirtyRegion = m_currentSettings.m_worldBounds;
-            m_dirtyRegion.AddAabb(m_requestedSettings.m_worldBounds);
+            m_dirtyRegion = m_cachedAreaBounds;
             m_terrainHeightDirty = true;
             m_terrainSurfacesDirty = true;
-            m_currentSettings.m_worldBounds = m_requestedSettings.m_worldBounds;
+            m_currentSettings.m_heightRange = m_requestedSettings.m_heightRange;
         }
```

Comment:
While this technically works, it should _probably_ use cached area bounds clamped to the old range, plus cached area bounds clamped to the new range, to get a slightly more accurate / constrained Z range in the case where one of the bounds is clamping a spawner's AABB.